backend/src/main/kotlin/org/loculus/backend/service/submission/FileMappingPreconditionValidator.kt

```diff
@@ -44,17 +44,6 @@ class FileMappingPreconditionValidator(
         return validateCategoriesMatchSchema(fileCategoriesFilesMap, allowedFileCategories, organism)
     }
 
-    fun validateCategoriesMatchProcessedSchema(
```

Comment:
@fhennig and @chaoran-chen I'm getting a bit confused with these changes - it seems like this is a rename of the functions added in https://github.com/loculus-project/loculus/pull/4385 where `validateCategoriesMatchProcessedSchema` has been renamed to `validateCategoriesMatchOutputSchema` and the error message has been changed. Otherwise the functions do the same thing - can we condense the functions instead?

Reply:
(I'll review the changes once they are in their own PRs: https://github.com/loculus-project/loculus/pull/4344/files#r2157417223)